### PR TITLE
[Messaging] Update attributes available for filtering

### PR DESCRIPTION
**Pull request by pwasowski2**
### Change Description ###

Related ACR - http://suprem.sec.samsung.net/jira/browse/TWDAPI-205 - makes it possible to use more attributes to filter Messages, MessageFolders and MessageConversations.
This commit adds the information about newly supported attributes to
the "Data Filtering and Sorting" guide.

The change adds HTML anchors to the tables with respective message filter attributes, so that it will be possible to link directly to them.

### API Changes ###

 - ACR: http://suprem.sec.samsung.net/jira/browse/TWDAPI-205

### CODE REVIEW

**Comment by huytaduc** [comment]
checkout the webpage https://docs.stg.tizen.org/staging/546

**Comment by pkosko** [comment]
Should not be merged now, need HQ acceptance.

**Comment by huytaduc** [comment]
checkout the webpage https://docs.stg.tizen.org/staging/546

**Comment by huytaduc** [comment]
checkout the webpage https://docs.stg.tizen.org/staging/546

File: docs/application/web/guides/data/data-filter.md [hunk 1]
```diff
@@ -330,15 +330,15 @@ The following table lists the filter types you can use with specific content att
 
 The following tables list the filter types you can use with specific message attributes in the methods of the [MessageStorage](../../api/latest/device_api/mobile/tizen/messaging.html#MessageStorage) interface.
 
-The following table lists the filter attributes related to the `findMessage()` method.
+The following table lists the attributes by which `Message` objects may be filtered.
 
-**Table: Filter attributes for finding messages**
+<span id="message-filters-table">**Table: Filter attributes for finding Messages**</span>
```
Comment (mijin85cho) [inline docs/application/web/guides/data/data-filter.md]: When you update this PR, please remove this span tag.  
Only **\*\*Table: Filter attributes for finding Messages\*\*** is enough.

File: docs/application/web/guides/data/data-filter.md [hunk 2]
```diff
@@ -351,44 +351,44 @@ The following table lists the filter attributes related to the `findMessage()` m
...
-| `messageStatus`  | No                         | No                               |
+| `inResponseTo`   | Yes                        | No                               |
+| `messageStatus`  | Yes                        | No                               |
 | `attachments`    | No                         | No                               |
 
-The following table lists the filter attributes related to the `findConversations()` method.
+The following table lists the attributes by which `MessageConversation` objects may be filtered.
 
-**Table: Filter attributes for finding conversations**
+<span id="conversation-filters-table">**Table: Filter attributes for finding conversations**</span>
```
Comment (mijin85cho) [inline docs/application/web/guides/data/data-filter.md]: When you update this PR, please remove this span tag.  
Only **\*\*Table: Filter attributes for finding conversations\*\*** is enough.

File: docs/application/web/guides/data/data-filter.md [hunk 3]
```diff
@@ -351,44 +351,44 @@ The following table lists the filter attributes related to the `findMessage()` m
...
-| `lastMessageId`  | No                         | No                               |
+| `cc`             | Yes                        | No                               |
+| `bcc`            | Yes                        | No                               |
+| `lastMessageId`  | Yes                        | No                               |
 
-The following table lists the filter attributes related to the `findFolders()` method.
+The following table lists the attributes by which `MessageFolder` objects may be filtered.
 
-**Table: Filter attributes for finding message folders**
+<span id="folder-filters-table">**Table: Filter attributes for finding folders**</span>
```
Comment (mijin85cho) [inline docs/application/web/guides/data/data-filter.md]: When you update this PR, please remove this span tag.  
Only **\*\*Table: Filter attributes for finding folders\*\*** is enough.

**Comment by mijin85cho** [comment]
> should not be merged until HQ acceptance

I'll ask the review to SRI-B after the HQ acceptance. So please let me know when to ask the review.

File: docs/application/web/guides/data/data-filter.md (hunk 1 above)
Comment (pwasowski2) [reply docs/application/web/guides/data/data-filter.md]: Our intention was to make it possible to the tables in this guide from Web API reference.

Currently a hyperlink may only point to _Messaging Filter Attributes_ - it has an _id="messaging-filter-attributes"_ attribute.

Added _<span id="message-filters-table">_  enables linking straight to the tables specific for _Message_ class.

So, can we preserve this _<span>_ tag, to enable linking straight to the table specific for _Message_?

Comment (mijin85cho) [reply docs/application/web/guides/data/data-filter.md]: If you want to create a link to a point which is not a title, you can use `<a name="message-filters-table"></a>`

For example, if you check web/guides/multimedia/media-controller.md file, 
you'll see "send_custom_commands" is inserted as below to jump to the one of list item.

image

Would you please use `<a name>` to link to the tables?

File: docs/application/web/guides/data/data-filter.md (hunk 2 above)
Comment (mijin85cho) [reply docs/application/web/guides/data/data-filter.md]: If you want to create a link to a point which is not a title, you can use `<a name="conversation-filters-table"></a>`

For example, if you check web/guides/multimedia/media-controller.md file, 
you'll see "send_custom_commands" is inserted as below to jump to the one of list item.

image

Would you please use `<a name>` to link to the tables?

File: docs/application/web/guides/data/data-filter.md (hunk 3 above)
Comment (mijin85cho) [reply docs/application/web/guides/data/data-filter.md]: If you want to create a link to a point which is not a title, you can use `<a name="folder-filters-table"></a>`

For example, if you check web/guides/multimedia/media-controller.md file, 
you'll see "send_custom_commands" is inserted as below to jump to the one of list item.

image

Would you please use `<a name>` to link to the tables?

File: docs/application/web/guides/data/data-filter.md (hunk 1 above)
Comment (pwasowski2) [reply docs/application/web/guides/data/data-filter.md]: I have applied the suggested change - anchors are now added with <a name="..."></a>.

File: docs/application/web/guides/data/data-filter.md (hunk 2 above)
Comment (pwasowski2) [reply docs/application/web/guides/data/data-filter.md]: I have applied the suggested change - anchors are now added with <a name="..."></a>.

File: docs/application/web/guides/data/data-filter.md (hunk 3 above)
Comment (pwasowski2) [reply docs/application/web/guides/data/data-filter.md]: I have applied the suggested change - anchors are now added with <a name="..."></a>.

**Comment by huytaduc** [comment]
checkout the webpage https://docs.stg.tizen.org/staging/546

**Comment by pwasowski2** [comment]
I have added an additional commit, discussing some ambiguous issues. This change was suggested in a face-to-face discussion with colleagues, who don't have an access to the review.

**Comment by journey2w** [comment]
======================<<< **WARNING: Coding Style** >>>=======================
docs/application/web/guides/data/data-filter.md: 142: ERROR: [BRC_M_FTN] open brace '{' following function declarations go on the next line

**Comment by pwasowski2** [comment]
I have closed this PR as we won't develop the features described in it.

**Pull request closed without merging**